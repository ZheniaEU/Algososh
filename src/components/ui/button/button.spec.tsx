import { fireEvent, render, screen } from "@testing-library/react"
import { Button } from "./button"

describe("Button tests", () => {

   describe("Button tests with snapshot", () => {

      it("button with text", () => {
         const { asFragment } = render(<Button text={"Кнопка с текстом"} />)
         expect(asFragment()).toMatchSnapshot()
      })

      it("button without text", () => {
         const { asFragment } = render(<Button />)
         expect(asFragment()).toMatchSnapshot()
      })

      it("button disabled", () => {
         const { asFragment } = render(<Button disabled={true} />)
         expect(asFragment()).toMatchSnapshot()
      })

      it("button with loading", () => {
         const { asFragment } = render(<Button isLoader={true} />)
         expect(asFragment()).toMatchSnapshot()
      })

   })

   it("test call callback", () => {
      const callback = jest.fn()
      render(<Button onClick={callback} />)
      fireEvent.click(screen.getByRole("button"))
      expect(callback).toBeCalledTimes(1)
   })

})
