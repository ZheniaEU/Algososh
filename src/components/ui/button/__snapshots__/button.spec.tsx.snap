// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests Button tests with snapshot button disabled 1`] = `
<DocumentFragment>
  <button
    class="text_type_button text_color_primary button undefined false "
    disabled=""
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</DocumentFragment>
`;

exports[`Button tests Button tests with snapshot button with loading 1`] = `
<DocumentFragment>
  <button
    class="text_type_button text_color_primary button undefined loader "
    disabled=""
    type="button"
  >
    <img
      alt="Загрузка."
      class="loader_icon"
      src="loader.svg"
    />
  </button>
</DocumentFragment>
`;

exports[`Button tests Button tests with snapshot button with text 1`] = `
<DocumentFragment>
  <button
    class="text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    >
      Кнопка с текстом
    </p>
  </button>
</DocumentFragment>
`;

exports[`Button tests Button tests with snapshot button without text 1`] = `
<DocumentFragment>
  <button
    class="text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</DocumentFragment>
`;

exports[`Button tests button disabled 1`] = `
<DocumentFragment>
  <button
    class="text_type_button text_color_primary button undefined false "
    disabled=""
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</DocumentFragment>
`;

exports[`Button tests button with loading 1`] = `
<DocumentFragment>
  <button
    class="text_type_button text_color_primary button undefined loader "
    disabled=""
    type="button"
  >
    <img
      alt="Загрузка."
      class="loader_icon"
      src="loader.svg"
    />
  </button>
</DocumentFragment>
`;

exports[`Button tests button with text 1`] = `
<DocumentFragment>
  <button
    class="text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    >
      Кнопка с текстом
    </p>
  </button>
</DocumentFragment>
`;

exports[`Button tests button without text 1`] = `
<DocumentFragment>
  <button
    class="text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</DocumentFragment>
`;

exports[`button tests button disabled 1`] = `
<DocumentFragment>
  <button
    class="text_type_button text_color_primary button undefined false "
    disabled=""
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</DocumentFragment>
`;

exports[`button tests button with loading 1`] = `
<DocumentFragment>
  <button
    class="text_type_button text_color_primary button undefined loader "
    disabled=""
    type="button"
  >
    <img
      alt="Загрузка."
      class="loader_icon"
      src="loader.svg"
    />
  </button>
</DocumentFragment>
`;

exports[`button tests button with text 1`] = `
<DocumentFragment>
  <button
    class="text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    >
      Кнопка с текстом
    </p>
  </button>
</DocumentFragment>
`;

exports[`button tests button without text 1`] = `
<DocumentFragment>
  <button
    class="text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</DocumentFragment>
`;
